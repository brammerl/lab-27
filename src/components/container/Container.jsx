import React, { Component } from 'react'
import ShortForm from '../short-form/ShortForm';
import DisplayUser from '../display-user/DisplayUser';
import DisplayRepos from '../display-repos/DisplayRepos';
import { getUserInfo } from '../../services/getUserInfo';
import { getRepos } from '../../services/getRepos';


export default class Container extends Component {
  state = {
    userName: '',
    userInfo: null,
    userRepos: null
  };

  handleChange = ({target}) => {
    this.setState({userName: target.value})
  }

  handleSubmit = event => {
    event.preventDefault();

    const { userName } = this.state;
    console.log(userName);

    Promise.all([
      getUserInfo(userName),
      getRepos(userName)
    ])
      .then(([userInfo, userRepos]) => this.setState({ userInfo, userRepos }))
  }
  render() {
    const {userName, userInfo, userRepos} = this.state;
    return (
      <>
        <ShortForm userName={userName} onChange={this.handleChange} onSubmit={this.handleSubmit}/>
        {
          userInfo ? <DisplayUser userInfo={userInfo}/> : <></>
        }
        {/* {
          userRepos ? 
            <DisplayRepos userRepos={userRepos}/>: <></>
        } */}
        
      </>
    )
  }
}
