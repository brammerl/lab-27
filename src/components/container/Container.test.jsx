import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container Component', () => {
  it('renders the container', () => {
    const wrapper = shallow(<Container/>)
    expect(wrapper).toMatchSnapshot();
  })
})