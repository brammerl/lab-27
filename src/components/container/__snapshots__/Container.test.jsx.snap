// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders the container 1`] = `
<Fragment>
  <ShortForm
    onChange={[Function]}
    onSubmit={[Function]}
    userName=""
  />
</Fragment>
`;
